### PR TITLE
Upgrade to strapi v4

### LINKED ISSUE #147: Upgrade Strapi to v4
This issue tracks the work to update Strapi v4.

>❗ Strapi v3 is in maintenance mode for 6 months and then it'll be not supported anymore (EOL date is May 30th, 2022)
https://github.com/strapi/strapi/issues/11726

## Known issues

- No published docker image for v4 yet

## To do

- [x] Bootstrap new backend folder with Strapi v4 cli
- [x] Create all models
- [x] Refactor seeding into a Strapi plugin
- [x] Refactor GraphQl queries
- [x] Refactor data fetching logic
- [x] Debug docker deploy

### PR DESCRIPTION
**Pull request by dhmacs**
This PR closes #147 and #157 

## Changelog

> ⚠️ Disclaimer for reviewers: Almost all changes under `/backend` folder have been auto generated by replacing the old project with the new one. I've been working mostly on:
> - `/backend/src/plugins/seed` directory
> - docker files

- Upgrades Strapi to v4
- Refactor Strapi seed plugin according to v4 API changes
- Updates Dockerfile
- Makes frontend use [Govinor](https://govinor.com) strapi branch deployment.

> **Govinor** is a custom build deployment platform for Strapi (but it could in practice deploy any docker compose file). This will give us separate branch preview urls for Strapi. You can read more [here](https://github.com/iFixit/govinor). 

## QA

1. Visit the Vercel PR preview
2. Verify that the website works like in the main branch (https://react-commerce.vercel.app/)
3. Verify that an instance of Strapi v4 is running at https://upgrade-to-strapi-v4.govinor.com/admin
4. Try to change some data on Strapi and verify that the site updates accordingly

### CODE REVIEW

**Comment by ardelato** [comment]
So far so good, waiting to gain permissions to `Govinor` to finish testing **step 3 & 4**

There were some differences in names and model information, but I am assuming that is from a difference in content between `Govinor` and the Strapi instance being used for the master branch?

### Steps Taken:
1. Compared this branch to master 
   - https://react-commerce-git-upgrade-to-strapi-v4-ifixit.vercel.app/collections/parts
   - https://react-commerce.vercel.app/collections/parts

- [x] Compared the Parts Model Behavior
Missing Models
- [x] Compared Filtering Behavior
Filtering Conditions
No Results Button
- [x] Ensured bugs that were previously seen were not present again
https://github.com/iFixit/react-commerce/pull/64#issuecomment-894476082
- [x] Compared Header bar behavior
- [x] Compared Footer behavior 
- [x] Compared mobile behavior

**Comment by dhmacs** [comment]
@ardelato 

> There were some differences in names and model information, but I am assuming that is from a difference in content between Govinor and the Strapi instance being used for the master branch?

What differences are you referring to?

**Comment by ardelato** [comment]
@dhmacs I noticed that not all the models were listed on this branch and that some of the model descriptions are different 

|Master|This Branch	|
|:---:|:---:|
|Master Models	|New Branch Model|


|Master - Amazon Kindle	| This Branch - Amazon Kindle	|
|:---:|:---:|
|Amazon Kindle Master|This Branch Amazon Kindle|

**Comment by dhmacs** [comment]
> @dhmacs I noticed that not all the models were listed on this branch and that some of the model descriptions are different

@ardelato Right, that is because each new Strapi branch deployment gets a bunch of seed data that I've generated. To get more data we'll need to update the seeds.

@sterlinghirsh you need also to update the integration that loads data into Strapi to v4 and make it work with the main branch ( https://main.govinor.com which we'll be created once we merge this PR).

**Comment by ardelato** [comment]
@dhmacs does that mean that although I am seeing `Samsung Laptop Parts` as a child of `parts` in Strapi, because the seed data has not been updated, its expected that `Samsung Laptop Parts` will not be seen on the `Parts` page?


|Strapi View	| Parts Page	|
|:---:|:---:|
|Children listed Strapi	|Actual Models listed	|

deploy_block ☁️  on that question👆 

---

Aside from that QA 👍 

## Steps Taken:
1. Continued the testing from https://github.com/iFixit/react-commerce/pull/158#issuecomment-1010513931
2. I confirmed that editing the `Description` of the `All Parts` item in the `ProductList` collection, was updating the content seen on the Vercel preview
Edited Description Strapi
Description Edited

**Comment by dhmacs** [comment]
> @dhmacs does that mean that although I am seeing `Samsung Laptop Parts` as a child of `parts` in Strapi, because the seed data has not been updated, its expected that `Samsung Laptop Parts` will not be seen on the `Parts` page?

@ardelato oh good catch! That's my fault, I missed that Strapi changed the defalt page size from 100 to 10, so that's why you see only 10 items. I'm going to specify the page size on the query 👍

**Comment by dhmacs** [comment]
@ardelato it's done ✅

**Comment by ardelato** [comment]
QA 👍  && un_deploy_block 🌻 

Confirmed that now all 38 children (models) are listed on the `parts` page. 

Children Listed

File: backend/config/admin.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = ({ env }) => ({
+  auth: {
+    secret: env('ADMIN_JWT_SECRET', 'd8ea0029965dfd06dfd92608b1ee0eff'),
```
Comment (sterlinghirsh) [inline backend/config/admin.js:3]: This is the default right? I know this question has come up a few times before.

**Comment by sterlinghirsh** [comment]
CR :zap: Looks good to me. I left a minor comment but I'm going to merge this since it's been sitting around for a long time.

File: backend/config/admin.js (hunk 1 above)
Comment (dhmacs) [reply backend/config/admin.js:3]: Yes that's autogenerated. I'll create a task to remove the default value so we are forced to set the env variable 👍

**Pull request merged**